doc/specs/Proto extensions-spec.md

```diff
@@ -0,0 +1,139 @@
+new profile objects from stubs that contain at least the following
+
+* A name (and this name _should not_ match the names of existing profiles as described above, because that would cause us to interpret this profile as modifying an existing one)
+* A commandline argument
+* A unique GUID - this is so we avoid conflicts if several different creators name their profile the same thing
+
+As in the case of the dynamic profile generator, if we create a profile that did not exist before (i.e. does not
+exist in the user settings), we need to add the profile to the user settings file and re-save that file. 
+
+#### Colour schemes
+
+Once again, we could run into the problem of several json stubs wanting to modify the same colour scheme. We
+will require a hierarchy here as well. Though since the 'owner' of a colour scheme is a lot more dubious, I wonder
+if we should only allow creation of new colour schemes for now and ignore requests to modify existing colour
+schemes. 
+
+### Location(s) of the json files
+
+For apps that are installed 'traditionally', their installer could simply add a json file to a specific local folder
+(specified by us) that we will look through every time we generate profiles. This is also where users can add
+their own json files for Terminal to generate profiles from. 
+
+For apps that are installed through something like the Microsoft Store, they should add their json stub(s) to
+an app extension. We will then obtain them during profile generation. 
+
+## UI/UX Design
+
+This feature will allow other installations a level of control over how their profiles look in Terminal. For example,
+if Ubuntu gets a new icon or a new font they can have those changes be reflected in Terminal users' Ubuntu profiles.
+
+Furthermore, this allows users an easy way to share profiles they have created - instead of needing to modify their
+settings file directly they could simply download a json file into a specific folder. 
+
+## Capabilities
+
+### Accessibility
+
+This change should not affect accessibility.
+
+### Security
+
+Opening a profile causes its commandline argument to be automatically run. Thus, if malicious modifications are made
+to existing profiles or new profiles with malicious commandline arguments are added, users could be tricked into running
+things they do not want to.
+
+### Reliability
+
+This should not affect reliability - most of what its doing is simply layering json which we already do. 
+
+### Compatibility
+
+Again, there should not be any issues here - the user settings can still be layered after this layering for the user
+to have the final say. 
+
+### Performance, Power, and Efficiency
+
+Looking through the additional json files could negatively impact startup time. 
+
+## Potential Issues
+
+Cases which would likely be frustrating:
+
+* An installer dumps a _lot_ of json files into the folder which we need to look through
+
+## Future considerations
+
+How would this affect the Settings UI? 
+

```

Comment:
@carlos-zamora, would like to hear your input here!

Reply:
I'm still a bit confused as to what changes we're making to the settings.json. So I'll hold off on answering this for now.

You may want to look at this for now: https://github.com/microsoft/terminal/blob/cinnamon/spec-settings-ui/doc/specs/%231564%20-%20Settings%20UI/design.md

We are planning on adding a "Marketplace" nav item. So it makes sense that this would go there. But we can probably hold off on a full design until after this spec gets merged in.

Reply:
We aren't making changes to the structure of settings.json - only the potential additions of more profiles similar to the way dynamic profiles are added. So I want to say that it probably wouldn't affect the settings UI but I will leave it to you to confirm!

Reply:
Looks like the only impact this would have is how we display "source" in the Settings UI. So I'll just keep this in mind as we go. No updates needed on your end :)